### ISSUE TITLE
PHP version 7.2

### ISSUE THREAD

**Issue by kyrenya** [tripal/tripal#719]
Hi, I'm about to install v3 from scratch. 
I will follow instructions here
https://tripal.readthedocs.io/en/latest/user_guide/install_tripal/server_setup.html

- is it ok to download the default php 7.2 (ubuntu 18.04) (sudo apt-get install php php-dev php-cli libapache2-mod-php)  or will I have compatibility issues? (I tried in a first attempt to downgrade to 7.1 but apparently I didnt do it very well, and then no further steps worked for me, thats why I'm restarting)

Thanks!

**Comment by shawnawsu** [comment]
Drupal 7 does not fully support php 7.2 so you might see errors or problems in terms of that. Please see this for more info: https://www.drupal.org/project/drupal/issues/2947772

**Comment by kyrenya** [comment]
ok! so for what I understand, I should apply that last patch (only one, not three) and fix it. I'll try and keep posted if i manage. Thanks!

**Comment by bradfordcondon** [comment]
hi @kyrenya : any follow up on this?  Did you go ahead with PHP 7.1 or 7.2?  

In terms of this issue, I think @shawnamspoor 's response more or less covers it, so if you have no other questions on this I'll close the issue.

**Comment by kyrenya** [comment]
Hi! I went on with PHP 7.2, tried to follow instructions in the link, but still I have the 
![image](https://user-images.githubusercontent.com/43243429/47431461-c3cdaa00-d793-11e8-8d36-0552abf70c2b.png)
But the warning doesnt bother me too much, so I moved on.. 
I tried patching with drupal-7-php-7-2-2947772-39.patch  ( I think that is the correct one, but if its other or I have to use several.. please let me know), but it FAILED at patching.

Thanks!

**Comment by bradfordcondon** [comment]
looking at  https://www.drupal.org/project/drupal/issues/2947772 ONE of those patches is the right one that lets drupal 7 support php 7.2  The one you linked seems like the right one to me (is it just me or are these drupal issues really hard to follow?  which is the final patch?  Did it ever resolve all errors? Shawna you are saying no, so kyrenya I think the answer is "proceed at your own risk with PHP 7.2).

**Comment by kyrenya** [comment]
thank god you also think its a bit hard.. XD XD

**Comment by kyrenya** [comment]
thanks, anyway, if you think its better to downgrade php to 7.1 or any other version..  I would appreciate any help with that.. I already tried by myself, and ended up with conflicts with the two versions, and the whole thing not working (dummy alert).

**Comment by bradfordcondon** [comment]
i just read that Drupal 7.61 officially supports PHP 7.2:  https://www.drupal.org/blog/the-end-of-php-5

**Comment by kyrenya** [comment]
Already updated to 7.61 thanks! 👍

**Comment by bradfordcondon** [comment]
does it resolve the problems you reported ie https://github.com/tripal/tripal/issues/719#issuecomment-432655610 ?

**Comment by kyrenya** [comment]
uhmm... the warnings still there
![image](https://user-images.githubusercontent.com/43243429/48339489-42fe2180-e660-11e8-92d6-5213ed6c5e2d.png)

**Comment by spficklin** [comment]
I can confirm that I get those same errors using PHP 7.2 with Drupal 7.61.  I think there must still be an issue with full support of PHP 7.2 and Drupal.    However, I have been using PHP 7.2 with Drupal and Tripal for many months and aside from the annoying warning messages it works just fine.

**Comment by kyrenya** [comment]
I also by now have only those warnings, if I have major issues I will let you know thanks :)

**Comment by spficklin** [comment]
Sometimes those warnings will appear for your users which is not great.  You can hide them by going to Admininstration > Configuration > Development > Logging and errors and clicking the "None" radio button.  Users will still see Drupal warnings/errors but these PHP warnings should go away.  It also states "It is recommended that sites running on production environments do not display any errors".   I believe the errors will still be logged, however.  I'm not 100% sure though.  

I'm closing this as it looks like things are resolved.  But please repost if there's more to discuss.

**Comment by kyrenya** [comment]
cool, good to know! thx!
